### PR TITLE
ServerSide Props + Styling Review + Incremental static regeneration + Cypress

### PR DESCRIPTION
**Pull request by AliLop**
ServerSide Props
Styling Review
Incremental static regeneration
Cypress

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (wilsonTheBallWhoCodes) [inline cypress/support/commands.js:1]: do we need this file?

File: src/_ProbesService.js [hunk 2]
```diff
@@ -26,3 +26,25 @@
 //     'https://www.thecocktaildb.com/api/json/v1/1/filter.php?c=Cocktail',
 //   );
 // }
+
+// /import { useEffect, useState } from 'react';
```
Comment (wilsonTheBallWhoCodes) [inline src/_ProbesService.js]: do we need this file?

File: src/components/Header.js [hunk 3]
```diff
@@ -4,9 +4,14 @@ import PropTypes from 'prop-types';
 function Header({ subtitle }) {
   return (
     <Container>
-      <div style={{ margin: '25px 0' }}>
-        <Typography variant="h4" color="textSecondary" align="center">
-          {subtitle}
+      <div style={{ margin: '15px 0' }} data-cy="header">
```
Comment (wilsonTheBallWhoCodes) [inline src/components/Header.js]: here, we can use `<Box my={2 or something}`. And alse set the fontWeight here too.

File: src/components/Layout/TopBar.js [hunk 4]
```diff
@@ -44,8 +44,8 @@ function TopBar() {
           </Link>
         </Box>
         <Box>
```
Comment (wilsonTheBallWhoCodes) [inline src/components/Layout/TopBar.js]: is this `Box` needed?

File: src/components/NavBar.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (router.pathname === '/') {
+    window = 'home';
+  } else if (router.pathname === '/about') {
+    window = 'about';
+  }
+
+  return (
+    <div style={{ display: 'flex', cursor: 'pointer' }}>
+      <Box>
```
Comment (wilsonTheBallWhoCodes) [inline src/components/NavBar.js]: is this `Box` needed?

File: src/components/NavBar.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        <Link href="/" passHref>
+          <Typography
+            style={{ paddingRight: '20px' }}
+            color={window === 'home' ? 'textSecondary' : 'secondary'}
+          >
+            Home
+          </Typography>
+        </Link>
+      </Box>
+      <Box>
```
Comment (wilsonTheBallWhoCodes) [inline src/components/NavBar.js]: is this `Box` needed?

File: src/pages/cocktail/[id]/index.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+import Header from '../../../components/Header';
+
+function Redirect({ to }) {
+  const router = useRouter();
+
+  useEffect(() => {
+    router.push(to);
+  }, [router, to]);
+
+  return null;
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/cocktail/[id]/index.js:23]: why you need to return `null` here?

File: src/pages/cocktail/[id]/index.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+              image={`${drinkData.strDrinkThumb}/preview`}
+            />
+            <CardContent>
+              <Typography
+                style={{ display: 'inline-block' }}
+                variant="h6"
+                color="primary"
+              >
+                Best served in:
+              </Typography>{' '}
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/cocktail/[id]/index.js:68]: are `{' '}` necessary?

File: src/pages/cocktail/[id]/index.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  const detailsById = await details.json();
+  return {
+    props: {
+      drinkData: detailsById.drinks[0],
+    },
+    revalidate: 10,
+  };
+}
+
+// Server Side
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/cocktail/[id]/index.js]: can we remove this?

File: src/pages/cocktail/[id]/index.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+  const { id } = params;
+  const details = await fetch(
+    `https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=${id}`,
+  );
+  const detailsById = await details.json();
+  return {
+    props: {
+      drinkData: detailsById.drinks[0],
+    },
+    revalidate: 10,
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/cocktail/[id]/index.js:122]: why 10? any specific reason?

File: src/pages/index.js [hunk 11]
```diff
@@ -42,12 +26,25 @@ export default function Home({ data }) {
   );
 }
 
-export async function getStaticProps() {
+// export async function getStaticProps() {
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/index.js]: can we remove this?

File: src/pages/cocktail/[id]/index.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+  const data = await res.json();
+  const paths = data.drinks.map((drink) => ({
+    params: { id: drink.idDrink.toString() },
+  }));
+  return { paths, fallback: true };
+}
+
+export async function getStaticProps({ params }) {
+  const { id } = params;
+  const details = await fetch(
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/cocktail/[id]/index.js]: let's abstract this http calls to a different folder/file, to make it independent of the implementation and easy to test.

File: src/pages/index.js [hunk 13]
```diff
@@ -42,12 +26,25 @@ export default function Home({ data }) {
...
+//     props: {
+//       data: data.drinks,
+//     },
+//     revalidate: 10,
+//   };
+// }
+
+// Server Side
+export async function getServerSideProps() {
   const res = await fetch(
```
Comment (wilsonTheBallWhoCodes) [inline src/pages/index.js]: let's abstract this http calls to a different folder/file, to make it independent of the implementation and easy to test.

File: src/_ProbesService.js (hunk 2 above)
Comment (AliLop) [reply src/_ProbesService.js]: will remove it

File: src/pages/cocktail/[id]/index.js (hunk 10 above)
Comment (AliLop) [reply src/pages/cocktail/[id]/index.js:122]: No specific reason just understanding how the Incremental Static Regeneration works and requires.

File: src/pages/cocktail/[id]/index.js (hunk 7 above)
Comment (AliLop) [reply src/pages/cocktail/[id]/index.js:23]: Needs return statement. No need for null specifically but "to render nothing, return null." and it works

File: src/pages/cocktail/[id]/index.js (hunk 8 above)
Comment (AliLop) [reply src/pages/cocktail/[id]/index.js:68]: empty space as in-line block to separate both strings

File: src/components/Layout/TopBar.js (hunk 4 above)
Comment (AliLop) [reply src/components/Layout/TopBar.js]: Nope

File: cypress/support/commands.js (hunk 1 above)
Comment (AliLop) [reply cypress/support/commands.js:1]: To run the cypress tests, apparently

**Pull request merged**